crates/jstzd/src/task/octez_client.rs

```diff
@@ -0,0 +1,152 @@
+use super::{endpoint::Endpoint, octez_node::OctezNode};
+use std::path::PathBuf;
+use tempfile::{tempdir, TempDir};
+
+pub struct OctezClientConfig {
+    binary_path: Option<PathBuf>,
+    // if None, use temp directory
+    base_dir: Option<PathBuf>,
+    endpoint: OctezNodeEndpoint,
+    disable_unsafe_disclaimer: bool,
+}
+
+impl OctezClientConfig {
+    pub fn new(endpoint: OctezNodeEndpoint) -> Self {
+        Self {
+            binary_path: None,
+            base_dir: None,
+            endpoint,
+            disable_unsafe_disclaimer: false,
+        }
+    }
+    pub fn set_path(mut self, path: PathBuf) -> Self {
+        self.binary_path = Some(path);
+        self
+    }
+    pub fn set_base_dir(mut self, base_dir: PathBuf) -> Self {
+        self.base_dir = Some(base_dir);
+        self
+    }
+    pub fn set_endpoint(mut self, endpoint: OctezNodeEndpoint) -> Self {
+        self.endpoint = endpoint;
+        self
+    }
+    pub fn set_disable_unsafe_disclaimer(
+        mut self,
+        disable_unsafe_disclaimer: bool,
+    ) -> Self {
+        self.disable_unsafe_disclaimer = disable_unsafe_disclaimer;
+        self
+    }
+    pub fn build(mut self) -> OctezClient {
+        let base_dir = self
+            .base_dir
+            .take()
+            .map_or(OctezClientDirectory::default(), OctezClientDirectory::Path);
+        OctezClient {
+            binary_path: self.binary_path.take(),
+            base_dir,
+            endpoint: self.endpoint,
+            disable_unsafe_disclaimer: self.disable_unsafe_disclaimer,
+        }
+    }
+}
+
+pub enum OctezNodeEndpoint {
+    RpcEndpoint(Endpoint),
+    Node(OctezNode),
+}
+
+impl From<Endpoint> for OctezNodeEndpoint {
+    fn from(endpoint: Endpoint) -> Self {
+        OctezNodeEndpoint::RpcEndpoint(endpoint)
+    }
+}
+impl From<OctezNode> for OctezNodeEndpoint {
+    fn from(node: OctezNode) -> Self {
+        OctezNodeEndpoint::Node(node)
+    }
+}
+
+pub enum OctezClientDirectory {
+    TempDir(TempDir),
+    Path(PathBuf),
+}
+
+impl Default for OctezClientDirectory {
+    fn default() -> Self {
+        OctezClientDirectory::TempDir(tempdir().expect("Failed to create tempdir"))
+    }
+}
+
+// TODO: remove later
+#[allow(dead_code)]
+pub struct OctezClient {
+    // if None, use the binary in $PATH
+    binary_path: Option<PathBuf>,
```

Comment:
I don't think this should be `Option`. The default path should have been set in `build` and passed to the client.

Reply:
yes i was looking into the design doc and thought about it. however i felt that it would be better to distinguish between a path to the binary of type `PathBuf` vs a command "octez-client" explicitly like this. wdyt @zcabter

Reply:
Agree with Huan that `binary_path` shouldn't be optional. 

No need to distinguish between custom binary path and "octez-client" as the shell will do right thing.

Reply:
done